apps/hash-frontend/src/components/hooks/use-entity-by-id.ts

```diff
@@ -35,7 +35,16 @@ export const useEntityById = ({
         ...zeroedGraphResolveDepths,
         ...graphResolveDepths,
         entityId,
-        includePermissions,
+        /**
+         * Setting `includePermissions` to `undefined` results in a "Invariant
+         * Violation: Invalid variable referenced in @include directive." runtime
+         * error in Apollo Client.
+         *
+         * @todo: consider making `includePermissions` required in the relevant queries.
+         */
+        ...(typeof includePermissions === "boolean"
+          ? { includePermissions }
+          : {}),
```

Comment:
@CiaranMn setting `includePermissions` to `undefined` in the query variables causes a `Invariant Violation: Invalid variable referenced in @include directive.` runtime error in the apollo client.

I'm not sure why this is, but wonder whether we should make the variable non-optional by removing the default value of `false`, so that the generated TS types don't allow us to cause this error accidentally.

Reply:
Addressed in https://github.com/hashintel/hash/pull/3462/commits/2ccdbfaba2a5a765a22b59703b1a3c67585e1d4c